code/modules/antagonists/eldritch_cult/knowledge.dm/rust_lore.dm

```diff
@@ -0,0 +1,248 @@
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_mark/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target,/mob/living))
+		var/mob/living/L = target
+		L.apply_status_effect(/datum/status_effect/eldritch/rust)
+
+/datum/eldritch_knowledge/rust_blade_upgrade
+	name = "Toxic blade"
+	gain_text = "Let the blade guide you through the flesh."
+	desc = "Your blade of choice will now add toxin to your enemies system."
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/spell/rust_wave)
+	banned_knowledge = list(/datum/eldritch_knowledge/ash_blade_upgrade,/datum/eldritch_knowledge/flesh_blade_upgrade)
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_blade_upgrade/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.reagents.add_reagent(/datum/reagent/toxin,5)
+
+/datum/eldritch_knowledge/spell/rust_wave
+	name = "Wave of Rust"
+	desc = "You can now send a projectile that converts an area into rust."
+	gain_text = "Messenger's of hope fear the rustbringer!"
+	cost = 1
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave
+	next_knowledge = list(/datum/eldritch_knowledge/rust_final,/datum/eldritch_knowledge/spell/blood_siphon,/datum/eldritch_knowledge/summon/rusty)
+	route = "Rust"
+
+/datum/eldritch_knowledge/armor
+	name = "Armorer's ritual"
+	desc = "You can now create eldritch armor using a table and a kitchen knife."
+	gain_text = "For I am the heir to the throne of doom."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/flesh_ghoul)
+	required_atoms = list(/obj/structure/table,/obj/item/kitchen/knife)
+	result_atoms = list(/obj/item/clothing/suit/hooded/cultrobes/eldritch)
+
+/datum/eldritch_knowledge/essence
+	name = "Priest's ritual"
+	desc = "You can now transmute a tank of water into a bottle of eldritch water."
+	gain_text = "This is an old recipe, i got it from an owl."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/spell/ashen_shift)
+	required_atoms = list(/obj/structure/reagent_dispensers/watertank)
+	result_atoms = list(/obj/item/reagent_containers/glass/beaker/eldritch)
+
+/datum/eldritch_knowledge/rust_final
+	name = "Rustbringer's Oath"
+	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual you will be granted a spell of incredible power."
+	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for Rustbringer has come!"
+	cost = 3
+	required_atoms = list(/mob/living/carbon/human)
+	route = "Rust"
+	var/finished = FALSE
+
+/datum/eldritch_knowledge/rust_final/recipe_snowflake_check(list/atoms, loc)
+	if(finished)
+		return FALSE
+	var/counter = 0
+	for(var/mob/living/carbon/human/H in atoms)
+		counter++
+	if(counter >= 3)
+		return TRUE
```

Comment:
say, can you use this as a way to just straight up delete the corpses of your victims while you're murderboning?

like, you can still sacrifice more people to this ritual after you sac the required 3 people to activate it, right?

Reply:
good catch